File: src/main/java/fi/aalto/cs/apluscourses/ui/module/ModulesDialog.java
package fi.aalto.cs.apluscourses.ui.module;

import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;

import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.DialogWrapper;
import fi.aalto.cs.apluscourses.intellij.toolwindows.APlusToolWindowFactory;
import javax.swing.Action;
import javax.swing.JComponent;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class ModulesDialog extends DialogWrapper {
  @NotNull
  private final ModulesView modulesView;

  /**
   * Constructor.
   */
  public ModulesDialog(@NotNull Project project) {
    super(project);

    this.modulesView = APlusToolWindowFactory.createModulesView(project);

    setModal(false);

    init();
    setSize(400, 600);
  }

  @Override
  protected Action @NotNull [] createActions() {
    return new Action[] {new DialogWrapperExitAction(getText("ui.close"), OK_EXIT_CODE)};
  }

  @Override
  @Nullable
  public JComponent getPreferredFocusedComponent() {
    return modulesView.moduleListView;
  }

  @Override
  protected @Nullable JComponent createCenterPanel() {
    return modulesView.getBasePanel();
  }
}

File: src/main/kotlin/fi/aalto/cs/apluscourses/ui/module/ModulesDialog.kt
package fi.aalto.cs.apluscourses.ui.module

import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.DialogWrapper
import fi.aalto.cs.apluscourses.MyBundle
import fi.aalto.cs.apluscourses.toolwindows.createModulesView
import javax.swing.Action
import javax.swing.JComponent

class ModulesDialog(project: Project) : DialogWrapper(project) {
//    private val modulesView = createModulesView(project)

    /**
     * Constructor.
     */
    init {
        isModal = false

        init()
        setSize(400, 600)
    }

    override fun createActions(): Array<Action> {
        return arrayOf(DialogWrapperExitAction(MyBundle.message("ui.close"), OK_EXIT_CODE))
    }

    override fun getPreferredFocusedComponent(): JComponent {
        TODO()//return modulesView.
    }

    override fun createCenterPanel(): JComponent? {
        return null
//        return modulesView.content
    }
}
